Foundation/NSOrderedSet.swift

```diff
@@ -46,67 +49,67 @@ open class NSOrderedSet : NSObject, NSCopying, NSMutableCopying, NSSecureCoding,
         }
     }
     
-    public required convenience init?(coder aDecoder: NSCoder) {
+    public required init?(coder aDecoder: NSCoder) {
         guard aDecoder.allowsKeyedCoding else {
             preconditionFailure("Unkeyed coding is unsupported.")
         }
+        _orderedStorage = NSMutableArray()
+        _storage = NSMutableSet()
         var idx = 0
-        var objects : [AnyObject] = []
         while aDecoder.containsValue(forKey: ("NS.object.\(idx)")) {
             guard let object = aDecoder.decodeObject(forKey: "NS.object.\(idx)") else {
                 return nil
             }
-            objects.append(object as! NSObject)
+            _orderedStorage.add(object)
+            _storage.add(object)
             idx += 1
         }
-        self.init(array: objects)
     }
     
     open var count: Int {
         return _storage.count
     }
 
     open func object(at idx: Int) -> Any {
-        return _SwiftValue.fetch(nonOptional: _orderedStorage[idx])
+        return _orderedStorage.object(at: idx)
     }
 
     open func index(of object: Any) -> Int {
-        return _orderedStorage.index(of: _SwiftValue.store(object)) ?? NSNotFound
+        return _orderedStorage.index(of: object)
     }
 
     public convenience override init() {
         self.init(objects: [], count: 0)
     }
 
     public init(objects: UnsafePointer<AnyObject>!, count cnt: Int) {
-        _storage = Set<NSObject>()
```

Comment:
According to https://developer.apple.com/documentation/foundation/nsorderedset the two init functions that take an `UnsafePointer` are:
`init(objects: UnsafePointer<AnyObject>?, count: Int)` and
`init(objects: UnsafePointer<AnyObject>, count: Int)`. This one needs to be fixed to match one of those, and also could you possibly add the other one?

Reply:
Absolutely - good catch